collect_app/src/main/java/org/odk/collect/android/formmanagement/downloaderror/FormsDownloadErrorActivity.kt

```diff
@@ -0,0 +1,29 @@
+package org.odk.collect.android.formmanagement.downloaderror
+
+import android.os.Bundle
+import androidx.appcompat.widget.Toolbar
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import org.odk.collect.android.R
+import org.odk.collect.android.activities.CollectAbstractActivity
+import org.odk.collect.android.utilities.TranslationHandler
+
+class FormsDownloadErrorActivity : CollectAbstractActivity() {
```

Comment:
I think we can drop the `FormsDownload` prefix from a few things here as this errors screen doesn't need to be specific to form downloading:

* `FormDownloadsErrorActivity` -> `ErrorListActivity`
* `FormsDownloadErrorAdapter` -> `ErrorAdapter`
* `FormDownloadErrorItem` -> `ErrorItem`

The supporting layouts would then change as well. One change would be that the logic of creating the `"ID: blah Version: blah"` string would live outside the adapter, but I think that's nice personally. In that way we separate all knowledge of forms out of this code which makes it resusable and also means it won't be effected by any changes we make to forms.

If that was the structure in place - a more generic "errors" UI - do you think it would convince you that this could live in a separate module?

Reply:
Ok I agree with that. I implemented those naming changes and even tried to move the classes to a separate module. The problem is that it requires a lot of changes. One of those classes is an activity (ErrorListActivity) so we will need to move things like our toolbar, maybe `CollectAbstractActivity`, `ThemeUtils` to `androidshared` in order to use it. I feel like I don't have enough time today and it would be even better to do that in a separate pr later because it will be a pretty big pr.

Reply:
Ah fair enough! I think we can probably move the Activity without moving `CollectAbstractActivity` and `ThemeUtils` but it would probably involve pulling a couple of helpers out and it'd be good for us to chat about where things like the toolbar live as you say. If I get a moment next week (once this is merged) I'll do a quick sketch of how we could structure it, and then we can chat about it when you get back 👍